xbmc/Application.cpp

```diff
@@ -4865,6 +4865,13 @@ bool CApplication::OnMessage(CGUIMessage& message)
       if (url.IsProtocol("plugin"))
         XFILE::CPluginDirectory::GetPluginResult(url.Get(), file);
 
+      // Don't queue if next media type is different from current one
+      if ((!file.IsVideo() && m_pPlayer->IsPlayingVideo()) ||
```

Comment:
I mean explicitly stopping playback. The OnSomething methods are called on events. You want to stop current player and open next list item on different player, right?